Update instructions for exit 0 (#3256)
cloudflare/cloudflare-docs: products/pages/src/content/framework-guides/deploy-anything.md

</TableLayout>

> Unlike many of our framework guides, the build command and build directory for your site are going to be completely custom. If you do not need a build step, input `exit 0` into the **Build command** field and specify a **Build output directory** where your application's content lives.

Reviewer: ```suggestion
Unlike many of the framework guides, the build command and build directory for your site are going to be completely custom. If you do not need a build step, leave the **Build command** field empty and specify a **Build output directory** . The build output directory is where your application's content lives.
```

> Unlike many of our framework guides, the build command and build directory for your site are going to be completely custom. If you do not need a build step, input  `exit 0` into the **Build command** field and then specify the **Build output directory**, your build command will run in the folder specified and you will not need to add the same directory as a path in the **Root directory**.

Reviewer: ```suggestion
Unlike many of our framework guides, the build command and build directory for your site are going to be completely custom. If you do not need a build step, input  `exit 0` into the **Build command** field and specify the **Build output directory**. Your build command will run in the folder specified and you will not need to add the same directory as a path in the **Root directory**.
```

Author: We need to mention that if you set the build output directory as  ./public for example and it's the same as the root. There is no need to have that twice;  because instead of building at `/public` Pages will look in `/public/public`

Reviewer 2: Not quite – they're serving different purposes. 

Made up example: if you have your site living in the `public` directory, then:

```
root: / (default)
build: public
result ==> /public ✅

root: /public 
build: /
result ==> /public ✅

root: /public
build: public
result ==> /public/public ❌
```

We don't want to recommend people change the root directory unless they _actually_ need the work (the build command) to run from that location. 

The pipeline workflow is essentially this:

```sh
git clone <repo> tmpdir
cd tmpdir
if (root != "/") cd <root>
run <build command>
deploy <build output dir>
```

So if you don't have a build command, you're just resolving a destination folder somehow (eg "/public"). There _are_ two ways to get to that destination, but the _more correct_ way is to avoid `root` changes when unnecessary.

Author: > We don't want to recommend people change the root directory unless they actually need the work (the build command) to run from that location. 

@lukeed  That's what I am saying "you will not need to add the same directory as a path in the **Root directory** "... 

This PR addresses the confusion in Issue 3148, because it's not specified in docs, people might do this: 

```
root: /public
build: public
result ==> /public/public ❌
```
Which will lead to the build failing. 

In the UI there is a prompt to tell you to leave it alone if your site isn't in a subdirectory.

Author: "Your project’s root directory, where Cloudflare runs the build command. If your site is not in a subdirectory, leave this path value empty."

Reviewer 3: I think @deadlypants1973 's suggestion is more accurate that @Ekwuno 's original suggestion.

> Unlike many of our framework guides, the build command and build directory for your site are going to be completely custom. If you do not need a build step, input  `exit 0` into the **Build command** field and then specify the **Build output directory**, your build command will run in the folder specified and you will not need to add the same directory as a path in the **Root directory**.

The build command will not run in the build output directory. It will run in the root.

I don't think it's worth specifying that the user shouldn't put the same directory as both the build output directory and the root just because one person tripped up on that. The UI is relatively clear that most people won't need to set it, and by bringing it up here, I honestly think it just muddies the waters.

![image](https://user-images.githubusercontent.com/8484333/151334736-af57a0b6-74fe-440f-8826-a717c320a62e.png)

Author: I still think this should be documented a little differently or at least feedback in the UI to stop this mistake. I am happy to land this PR right now, and we can revisit at a later time.
